packages/core/src/tracing/instrumentation/messaging/kafkaJs.js

```diff
@@ -236,24 +246,30 @@ function instrumentedEachMessage(originalEachMessage) {
     }
 
     let traceId;
+    let longTraceId;
     let parentSpanId;
     let level;
     if (message && message.headers) {
       // Look for the the newer string header format first.
-      if (message.headers[constants.kafkaTraceIdHeaderNameString]) {
-        traceId = String(message.headers[constants.kafkaTraceIdHeaderNameString]);
+      if (message.headers[constants.kafkaTraceIdHeaderName]) {
+        traceId = String(message.headers[constants.kafkaTraceIdHeaderName]);
+        if (traceId) {
+          const limited = limitTraceId({ traceId });
```

Comment:
Note: Limiting the incoming trace ID here to 16 characters is one crucial change of this PR.